### PR TITLE
Get the most recent Podling Name Search

### PR DESCRIPTION
**Pull request by dave2wave**
Untested

### CODE REVIEW

File: lib/whimsy/asf/podling.rb [hunk 1]
```diff
@@ -432,6 +432,8 @@ def self.namesearch
         resolution = issue['fields']['resolution']
         resolution = resolution ? resolution['name'] : 'Unresolved'
         [name, {issue: issue['key'], resolution: resolution}]
+        # we want the podling name search with the largest key
+        break
```
Comment (rubys) [inline lib/whimsy/asf/podling.rb:436]: There are multiple problems with this:

 * This loop is intended to return a list of ALL resolutions for all names, which is converted later to a hash by name.  A break statement would stop at the first resolution for any name.
* blocks return the value of the last statement in the block.  As that last statement is a break, the end result is that `issues` will be a `nil`.

The `to_h` call below will convert the array to a hash, and the behavior of the conversion is that the last one with any given key wins.  The most expedient solution is to add `.reverse` prior to the call to `.to_h`.  A more comprehensive solution would be replace the `sort_by` to a `group_by` followed by code that selects the entry with the largest issue number in every group.

**Comment by sebbASF** [comment]
The code currently does not handle case differences well.
If the customfield_12310520 does not contain any upper-case letters, the name is derived from the title instead.
However no such restriction applies to the extracted title.

It might be better to convert all names to lower-case for matching purposes.
I don't think we should allow two podlings with names that differ only in case, but I don't see why a podling should not have a lower-case only name.

**Comment by sebbASF** [comment]
Perhaps customfield_12310520 should be checked against an RE of the form /^[a-zA-Z0-9 ]+$/

**Comment by sebbASF** [comment]
Also it occurs to me that the most recent key number may not be the most appropriate

**Comment by sebbASF** [comment]
Also, it looks like the customfield_12310521, i.e. 'Proposed Name' might be more suitable as the source of the podling name.

See for example: https://issues.apache.org/jira/browse/PODLINGNAMESEARCH-175
This is currently listed under 'MesaTEE' instead of 'TeaClave'

**Comment by rubys** [comment]
The purpose of this source file is to produce the following results: https://whimsy.apache.org/board/agenda/json/podlingnamesearch ; should names become canonicalized (for example, by lowercasing), the following file would also need to change: https://github.com/apache/whimsy/blob/master/www/board/agenda/views/elements/pns.rb

**Comment by sebbASF** [comment]
Maybe some work is needed by the Incubator to ensure that the correct name is always provided in the JIRA.
The Proposed Name field is probably the most suitable.

I think it may also need agreement with the person signing off the JIRA.
They would need to sign it off based on the appropriate field, not any variations that might be present in the title or elsewhere. Perhaps the sign-off comment should include the name explicitly, rather than by implication.

**Comment by dave2wave** [comment]
Thanks Sam for your comments.

Podling Name Search approval is the responsibility of VP, Brand. The Board has lately made it a requirement for all TLPs. Petri had to do one.

I do think that fixing things in JIRA is the proper approach. That will need to be discussed on trademarks@. Once done whimsy can adjust the query and consider separating this information from podlings.rb

Regards,
Dave

**Comment by sebbASF** [comment]
I did a check, and the contents of Proposed Name match much better with the Title field.

There is one instance where this is not the case; this is because of a typo in the Proposed Name field:
https://issues.apache.org/jira/browse/PODLINGNAMESEARCH-26

So I think we should change to using the Proposed Name field for a start.

**Pull request closed without merging**